### ISSUE TITLE
Better handling of groceries according to different supermarkets

### ISSUE THREAD

**Issue by update-freak** [TandoorRecipes/recipes#2378]
### Is your feature request related to a problem? Please describe.

Hello developer,

for me it's not that intuitive to assign a groceries to a specific supermarket, e.g. buy egg in supermarket xy.
I weekly check the offers and buy the groceries in different supermarkets where the price is the lowest.
Currently I cread the shopping list in that way that I add the supermarket in brackets after the groceries, e.g. milk (ALDI), egg (LIDL)

### Describe the solution you'd like

Maybe it would be an idea that the supermarket can be selected (and also the shopping category) when adding one item to the shopping list.
When I'm in the supermarket I will use the app https://github.com/phantomate/Untare to check the list and delete added groceries .

**Comment by vabene1111** [comment]
to be honest i do not understand what you are doing and why, but its probably due to the supermakret features not being very intuitive

- you can assign categories to foods
- you can assign categories to supermarkets
- you can filter that only categories assigned to your supermarket are shown 

it is planned to add a way to "split" the shopping list into multiple lists at some point so you could tag one with "lidl" and one with "aldi" for example but thats likely as much differentiation as will be added. 

Does this make sense ?

**Comment by update-freak** [comment]
ok, no problem.
For me it would be an improvement to have the option to select the supermarket when I add foods to the shopping list.
For example like this:
![supermarkt](https://user-images.githubusercontent.com/40196995/227719304-a5d994e6-27b1-4b94-8bfc-97255f88d82a.png)

When I'm in the supermarket ALDI, I want to do the following:
1. Filter my food for the supermarket ALDI
2. categorize the food by categories -> to start e.g. with vegetables
3. after I added everything from ALDI, I want to go the next supermarket LIDL and then do the same -> filter for LIDL and then categorize by categories

you can filter that only categories assigned to your supermarket are shown --> ok. Thats good, but for my example the category "vegetables" is necessary for both supermarkets (ALDI and LIDL), because in the supermarket ALDI e.g. I want to buy the cucumber, because it's on offer, and next time it is cheaper in the supermarket LIDL.
This means that for me it's not that intuitive to assign categories to supermarkets, because in that example you see that the category "vegetables" is valid for both supermarkets. Maybe it markes more sense to have it like that:

- assign categories to foods -> to have the overview in a specific market -> to start with e.g. vegetables
- assign foods food to supermarkets -> because a whole categorie might not that useful. Think about weekly offers for different supermarkets

With the tag-feature my problem/process can be also solved, but I think the above proposal (assign foods food to supermarkets) make more sense for me (but I'm not sure what are the consequences in programming/implementation).

I hope this helps to unterstand my process of shopping according to weekly offers and the need to add food to supermarkets.
Please let me know if something it not clear. Thanks a lot!

**Comment by vabene1111** [comment]
I understand what you me. 

Adding it with a tag based approach solves two problems, for one yours but also the one where you want to add entries to lets say a list a friend should bring or something that you want to note you wanna buy but dont want to buy just yet, so I think i will go that route. 

This is not a high priority thing but something that has bugged me for a while, closing this issue as the tag thing is already planned but I will keep your wish in mind when doing this.

**Comment by update-freak** [comment]
So you will add a tag on the location where the supermarket field is, right? But I think then you have the supermarket double -> you have the association from the groceries to supermarkt categories and in addition to that the label when there is a offer of one specific grocery in one special supermarket.
(Would then also good to have an autocomplete with the existing supermarkets when I tip the name).
My proposal:
![proposal](https://github.com/TandoorRecipes/recipes/assets/40196995/c860a1fa-727f-4d13-bfea-9598be8e0d98)

**Comment by Crumar** [comment]
I tried to request the same and got redirected here, and fully agree with @update-freak.
The current assignment does not make sense in my mind and is a bit hidden. I just tried Tandoor freshly to see if it can replace AnyList, for they refuse to implement an API.
AnyList does this exactly this way and I find this very intuitive. If its unassigned, its available on all stores. If it has stores assigned, its not visible when in another store.
Categories help with not running in circles within any given store, but I still understand the possibility to exclude certain categories from stores entirely, e.g. I'd usually not shop for gardening supplies at Aldi.
